grid/social-media/posts-lookup/maps/twitter.suma

```diff
@@ -155,32 +161,90 @@ operation MapMediaAttachment {
   return result
 }
 
+//Twitter rate limits: https://developer.twitter.com/en/docs/twitter-api/rate-limits
+operation MapRateLimit {
+  totalRequestsHeader = args.headers['x-rate-limit-limit']
+  remainingRequestsHeader = args.headers['x-rate-limit-remaining']
+  resetTimestampHeader = args.headers['x-rate-limit-reset']
+
+  totalRequests = undefined
+  remainingRequests = undefined
+  remainingRequestsPercentage = undefined
+  resetTimestamp = undefined
+
+  set if(totalRequestsHeader && parseInt(totalRequestsHeader, 10) !== 'NaN') {
+    totalRequests = parseInt(totalRequestsHeader, 10)
+  }
+
+  set if(remainingRequestsHeader && parseInt(remainingRequestsHeader, 10) !== 'NaN') {
+    remainingRequests = parseInt(remainingRequestsHeader, 10)
+  }
+
+  set if(totalRequests > 0 && remainingRequests !== undefined) {
+    remainingRequestsPercentage = remainingRequests / totalRequests * 100
+  }
+
+  set if(resetTimestampHeader && parseInt(resetTimestampHeader, 10) !== 'NaN') {
+    resetTimestamp = parseInt(resetTimestampHeader, 10)
+  }  
```

Comment:
I am thinking about this duplicated `parseInt`. How about parsing the headers directly and then only check for NaN? I _think_ `parseInt` never throws, it just gives you NaN or a garbage number.

```
totalRequestsHeader = args.headers['x-rate-limit-limit']
// ...
set if(totalRequestsHeader !== 'NaN') {
  totalRequests = totalRequestsHeader
}
```

Also not sure if NaN is handled correctly; can we use [Number.isNaN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN) or straight `isNaN` instead?

Reply:
Good points 👍

Reply:
To be honest I did not test the NaN edge case. I would like to have ability to write unit tests for Comlink -> mock inputs.

Reply:
fixed